libraries/AP_TECS/AP_TECS.cpp

```diff
@@ -470,11 +470,20 @@ void AP_TECS::_update_speed_demand(void)
     // Constrain speed demand, taking into account the load factor
     _TAS_dem = constrain_float(_TAS_dem, _TASmin, _TASmax);
 
+    // Determine the true cruising airspeed (m/s)
+    _TAScruise = 0.01f * (float)aparm.airspeed_cruise_cm * _ahrs.get_EAS2TAS();
+
     // calculate velocity rate limits based on physical performance limits
     // provision to use a different rate limit if bad descent or underspeed condition exists
-    // Use 50% of maximum energy rate to allow margin for total energy contgroller
+    // Use 50% of maximum energy rate on gain, 90% on dissipation to allow margin for total energy controller
     const float velRateMax = 0.5f * _STEdot_max / _TAS_state;
-    const float velRateMin = 0.5f * _STEdot_min / _TAS_state;
+    // Maximum permissible rate of deceleration value at max airspeed
+    const float velRateNegMax = 0.9f * _STEdot_neg_max / _TASmax;
+    // Maximum permissible rate of deceleration value at cruise speed
+    const float velRateNegCruise = 0.9f * _STEdot_min / _TAScruise;
+    // Linear interpolation between velocity rate at cruise and max speeds
+    const float velRateFinal = (((velRateNegMax - velRateNegCruise) / (_TASmax - _TAScruise)) * _TAS_state) + (velRateNegCruise - ((velRateNegMax - velRateNegCruise) / (_TASmax - _TAScruise)) * _TAScruise);
+    const float velRateMin = constrain_float(velRateFinal, velRateNegMax, velRateNegCruise);
```

Comment:
```suggestion
const float velRateMin = linear_interpolate(velRateNegMax, velRateNegCruise, _TAS_state, _TASmax, TAScruise);
```
I think you can just use our build in linear interpolate function?

Reply:
Yup, good call.
```cpp
const float velRateMin = constrain_float(linear_interpolate(velRateNegMax, velRateNegCruise, _TAS_state, _TASmax, TAScruise), velRateNegMax, velRateNegCruise);
```

Reply:
The `linear_interpolate` function already does the constrain.

Reply:
Updated.